tests/test_walk_utils.py

```
@@ -0,0 +1,115 @@
+def test_missing_columns():
+    df = make_valid_walk_df().drop(columns=['cadence'])
+    with pytest.raises((KeyError, ValueError)):
+        walk_utils.walk_block_segments(
+            df, 'is_walk', 'pace_min_per_km', 'cadence', 140
+        )
+
+# 2. Input validation: invalid types
+def test_invalid_types():
+    df = make_valid_walk_df()
+    df['cadence'] = 'bad'  # string instead of numeric
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)